examples/eks-auto-mode/index.ts

```diff
@@ -0,0 +1,139 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+import { SubnetType } from "@pulumi/awsx/ec2";
+
+const config = new pulumi.Config();
+const clusterName = config.require("clusterName");
+
+const eksVpc = new awsx.ec2.Vpc("eks-auto-mode", {
+    enableDnsHostnames: true,
+    cidrBlock: "10.0.0.0/16",
+    subnetSpecs: [
+        // Necessary tags for EKS Auto Mode to identify the subnets for the load balancers.
+        // See: https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/deploy/subnet_discovery/
+        {type: SubnetType.Public, tags: {[`kubernetes.io/cluster/${clusterName}`]: "shared", "kubernetes.io/role/elb": "1"}},
+        {type: SubnetType.Private, tags: {[`kubernetes.io/cluster/${clusterName}`]: "shared", "kubernetes.io/role/internal-elb": "1"}},
+    ],
+    subnetStrategy: "Auto"
+});
+
+const cluster = new eks.Cluster("eks-auto-mode", {
+    name: clusterName,
+    // EKS Auto Mode requires Access Entries, use either the `Api` or `ApiAndConfigMap` authentication mode.
+    authenticationMode: eks.AuthenticationMode.Api,
+    skipDefaultNodeGroup: true,
```

Comment:
All these defaults that we have to flip, are they the only option when using autoMode or can you still set `skipDefaultNodeGroup` to `false`?

Can we make different defaults when autoMode is enabled so the configuration is less verbose? As it is not it looks like enabling "autoMode" required _more_ configuration from the user.

e.g. can using autoMode look like this?

```ts
const cluster = new eks.Cluster("eks-auto-mode", {
    name: clusterName,
    vpcId: eksVpc.vpcId,
    publicSubnetIds: eksVpc.publicSubnetIds,
    privateSubnetIds: eksVpc.privateSubnetIds,
    autoMode: {
        enabled: true,
    }
});
```

Reply:
I updated the defaults for `skipDefaultNodeGroup` and `skipDefaultSecurityGroups`. I intentionally didn't update the defaults for the addons. While kube-proxy and vpc-cni are not necessarily required for Auto Mode, there's no downside in creating them and users need them if they add other node groups.

Reply:
While it is possible to have both Auto Mode and EKS managed node groups in the same cluster - the reality is that its highly unlikely. I don't know if it breaks your current user contract but it would be better to not deploy any addons by default when Auto Mode is enabled

Reply:
The downside to this approach is that users who are not fully versed on Auto Mode may view this as a bug or an EKS issue when the VPC CNI daemonst fails to create any pods, nor do any of their VPC CNI configurations show up

Reply:
It does break the user contract to some extent. Users of the provider expect it to manage the addons for them unless they opt out of it.
But the VPC CNI configuration not applying here is a good point, I didn't know about that! I think in that case it's better to opt for defaulting to false and letting users decide.

Reply:
Thanks @bryantbiggs for all the insights!

Reply:
I work with the EKS service team and manage the EKS Terraform module - I'm happy to have a quick chat sometime to answer any questions and provide some insights on what we commonly see folks doing within their EKS provisioning. Just send me a message brybiggs@amazon.com and we can set something up if interested